### PR TITLE
[Tools] Add the onnx-lsp-server for ONNXDialect and KrnlDialect

### PR DESCRIPTION
**Pull request by L19h7n1nG**
Incorporate the LSP server for ONNXDialect and KrnlDialect, along with valuable MLIR and LLVM dialects to enhance functionality

### CODE REVIEW

**Comment by tungld** [comment]
@jenkins-droid test this please

File: src/Tools/onnx-lsp-server/onnx-lsp-server.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  registry.insert<func::FuncDialect>();
+  registry.insert<vector::VectorDialect>();
+  registry.insert<shape::ShapeDialect>();
+  registry.insert<math::MathDialect>();
+  registry.insert<memref::MemRefDialect>();
+  registry.insert<cf::ControlFlowDialect>();
+  registry.insert<mlir::omp::OpenMPDialect>();
+
+  registry.insert<mlir::ONNXDialect>();
+  registry.insert<mlir::KrnlDialect>();
```
Comment (chentong319) [inline src/Tools/onnx-lsp-server/onnx-lsp-server.cpp:29]: Is it possible to reuse the code in CompilerDialect.cpp to keep the consistence in future?

Comment (L19h7n1nG) [reply src/Tools/onnx-lsp-server/onnx-lsp-server.cpp:29]: sure, i will reuse the code in ```src/Compiler/CompilerDialects.cpp``` and ```src/Compiler/CompilerOptions.cpp```

**Review by chentong319 — APPROVED** [review]
LGTM!

**Comment by chentong319** [comment]
@jenkins-droid test this please

**Comment by chentong319** [comment]
Please fix the format issue. You can either run the clang-format tool or modify the source code according the error message. It seems to me that the error here is just extra space or empty line.

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by L19h7n1nG** [comment]
Could we merge this pr into the main branch? It appears that there haven't been any changes to the compiler's pipeline, so additional regression tests might not be necessary

**Pull request merged**